src/schema/v2/partner.ts

```diff
@@ -284,6 +292,87 @@ export const PartnerType = new GraphQLObjectType<any, ResolverContext>({
           )
         },
       },
+      partnerArtistArtworksConnection: {
```

Comment:
If we haven't had a recommended pattern, this could be a good opportunity to establish one in Metaphysics for querying data based on different roles/authorization. Here are my 0.2 cents:

For example, this new field and the [existing one](https://github.com/artsy/metaphysics/blob/483f4fdd95f8b5efc5e089b1a2488df880999b68/src/schema/v2/partner_artist.ts#L127-L162) could be confusing:

```graphql
{
  partnerArtist(...) {
    artworksConnection # public, returns public/less data
  }
  partner(...) {
    partnerArtistArtworksConnection # requires authorized partner or admin, returns more data
  }
}
```

They both get artworks for a partner artist, but authorize differently and return different data. It's not immediately clear which one to use for a certain scenario. It would be great we have a consistent pattern to make our future life easier. I don't have a good answer but naively some options:

- **Use a single field for multiple purposes**. This [`artworksConnection` under partner](https://github.com/artsy/metaphysics/blob/62c985ab60f8ce348e53342571855691de0eb153/src/schema/v2/partner.ts#L287-L371) is an example. I think this is not ideal. The logic is complex. The arguments overlap but not always applicable. There are also more round trips.
- **Use a separate field for each use case**. This might be closer to this PR. I think it's cleaner implementation, and potentially can be less confusing to developers if, say, we name the fields consistently in certain way and/or write a clear field description.
- **Use a root field**. The [`me` root field](https://github.com/artsy/metaphysics/blob/604043ac24afbe724a10202e8dde88441ef4f142/src/schema/v2/schema.ts#L222) is clear that one has to be authorized as herself/himself to access her/his data in the me type. I wonder if we can borrow similar concept for accessing partner owned data.

I don't think this should be blocking. I thought it's a good timing to discuss and come up with a pattern for partner use cases in MP.

Reply:
I realized the example above wasn't accurate. The `partnerArtist { artworksConnection }` actually returns [`partner_artist_artworks`](https://github.com/artsy/metaphysics/blob/0e7202ed5899e9dee465ddded88950138ee7e5f3/src/lib/loaders/loaders_without_authentication/gravity.ts#L122) rather than `artworks`. But the idea should be the same.

Reply:
Thanks Chung Yi! after looking around a bit more it actually seems like most places are using the same field in most cases, e.g. your option 1 and when you hit the endpoint in an authenticated state it uses the authenticated loader and returns more data depending on your access. I agree this is not the most intuitive but since it seems like that is the standard I think I should stick with that, for this case it looks like I _can_ actually reuse the existing artworksConnection and pass some other args e.g. `published: false` So I've removed this extra connection and added some extra args and an authenticated loader to the existing field. 

For me it would make sense to have a separate field if the args for the authenticated request were significantly different or the authenticated request had to use a different endpoint. Let me know what you think!